src/m4/jtr_x86_logic.m4

```diff
@@ -244,13 +258,16 @@ if test "x$simd" = xyes; then
 
   AS_IF([test "x$CPU_NOTFOUND" = x0],
   [
-  CC="$CC_BACKUP -mavx2"
-  AC_MSG_CHECKING([for AVX2])
+  CFLAGS="$CFLAGS_BACKUP -mavx2 -P $EXTRA_AS_FLAGS $CPPFLAGS $CFLAGS_EXTRA $CPUID_ASM"
```

Comment:
You missed the (e.g.) `-DCPU_REQ_AVX2 -DCPU_REQ`, no? It is required by the ASM code.

Reply:
No, I dropped it because it's added by arch.h

Reply:
Same command line works perfectly fine up to AVX2, then starts returning "true" for AVX512* even though I fixed mic.h

Reply:
Thinking aloud:
- you need -DCPU_REQ_SSE2 when testing SSE2
- you need -DCPU_REQ_SSE3 when testing SSE3
- etc.

Are you sure it is changing as needed. If not, you are always testing SSE2.

Reply:
That's not it. I'm running AVX. It returns true for SSSE3, SSE4.1, AVX and then returns false for XOP and AVX2. While debugging I let it continue trying AVX512* anyway - and then it returns true for both types of AVX512.

When doing the f***n exact same thing with test_SIMD.c, it returns false for AVX512.

Reply:
That is interesting. I'll try to see if I can see a bug.

Reply:
I pushed a new version but it only dropped the `#include`s becuase they're not needed.

Reply:
I'm taking a nap, TTFN. Hopefully I'll know the answer when I wake up.

Reply:
There is a typo in your code.